containerResolver panic

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0x1996f28]

goroutine 1161 [running]:
github.com/matrixorigin/matrixcube/raftstore.(*store).containerResolver(0x1d09720, 0xc00a8fa070)
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/raftstore/store.go:754 +0x48
github.com/matrixorigin/matrixcube/transport.(*Transport).resolve(0xc00a17d200, 0x0, 0x2)
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/transport/transport.go:419 +0xf1
github.com/matrixorigin/matrixcube/transport.(*Transport).Send(_, {0x2, 0x0, {0x20, 0x1f, 0x1, 0x0, {}, {0x0, 0x0, ...}, ...}, ...})
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/transport/transport.go:239 +0x12a
github.com/matrixorigin/matrixcube/raftstore.(*replica).sendRaftMessage(_, {0x4, 0x3, 0x20, 0x2, 0x0, 0x7, {0x0, 0x0, 0x0}, ...})
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/raftstore/replica_event_raft_ready.go:288 +0x4d5
github.com/matrixorigin/matrixcube/raftstore.(*replica).sendMessage(_, {0x4, 0x3, 0x20, 0x2, 0x0, 0x7, {0x0, 0x0, 0x0}, ...})
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/raftstore/replica_event_raft_ready.go:234 +0x65
github.com/matrixorigin/matrixcube/raftstore.(*replica).send(0x0, {0xc009937b00, 0x1, 0x7}, 0x0)
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/raftstore/replica_event_raft_ready.go:228 +0x1d9
github.com/matrixorigin/matrixcube/raftstore.(*replica).sendRaftMessages(...)
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/raftstore/replica_event_raft_ready.go:216
github.com/matrixorigin/matrixcube/raftstore.(*replica).processReady(_, {0x0, {0x2, 0x0, 0x7}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/raftstore/replica_event_raft_ready.go:64 +0x21a
github.com/matrixorigin/matrixcube/raftstore.(*replica).handleRaftReady(0xc0006fb800, 0xc00cbe6000)
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/raftstore/replica_event_raft_ready.go:37 +0x134
github.com/matrixorigin/matrixcube/raftstore.(*replica).handleEvent(0xc0006fb800, 0x0)
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/raftstore/replica_event_loop.go:224 +0x1c5
github.com/matrixorigin/matrixcube/raftstore.(*replicaWorker).handleEvent(0xc00c2d67b0, {0x21c21c8, 0xc0006fb800})
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/raftstore/worker_pool.go:109 +0x5f
github.com/matrixorigin/matrixcube/raftstore.(*replicaWorker).workerMain(0xc00c2d67b0)
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/raftstore/worker_pool.go:86 +0xc5
github.com/matrixorigin/matrixcube/raftstore.newReplicaWorker.func1()
	/root/matrixone-1/vendor/github.com/matrixorigin/matrixcube/raftstore/worker_pool.go:72 +0x1d
github.com/lni/goutils/syncutil.(*Stopper).runWorker.func1()
	/root/matrixone-1/vendor/github.com/lni/goutils/syncutil/stopper.go:79 +0xd2
created by github.com/lni/goutils/syncutil.(*Stopper).runWorker
	/root/matrixone-1/vendor/github.com/lni/goutils/syncutil/stopper.go:74 +0xef
FAIL	github.com/matrixorigin/matrixone/pkg/vm/driver/test	11.528s
FAIL

```

--- a/raftstore/store.go
+++ b/raftstore/store.go
@@ -751,6 +751,9 @@ func (s *store) containerResolver(storeID uint64) (string, error) {
 	if err != nil {
 		return "", err
 	}
+	if container == nil {
+		return "", nil
+	}
 	return container.ShardAddr(), nil
 }
 

--- a/transport/transport.go
+++ b/transport/transport.go
@@ -42,6 +42,7 @@ import (
 	"go.etcd.io/etcd/raft/v3/raftpb"
 	"go.uber.org/zap"
 
+	"github.com/matrixorigin/matrixcube/components/log"
 	"github.com/matrixorigin/matrixcube/pb/meta"
 	"github.com/matrixorigin/matrixcube/snapshot"
 	"github.com/matrixorigin/matrixcube/vfs"
@@ -167,7 +168,7 @@ func NewTransport(logger *zap.Logger, addr string,
 	dir snapshot.SnapshotDirFunc,
 	resolver ContainerResolver, fs vfs.FS) *Transport {
 	t := &Transport{
-		logger:         logger,
+		logger:         log.Adjust(logger),
 		storeID:        storeID,
 		handler:        handler,
 		unreachable:    unreachable,
@@ -417,7 +418,7 @@ func (t *Transport) resolve(storeID uint64, shardID uint64) (targetInfo, bool) {
 	}
 
 	addr, err := t.resolver(storeID)
-	if err != nil {
+	if err != nil || addr == "" {
 		t.logger.Error("failed to resolve store addr",
 			zap.Error(err))
 		return targetInfo{}, false

--- a/transport/transport_test.go
+++ b/transport/transport_test.go
@@ -30,11 +30,17 @@
 package transport
 
 import (
+	"errors"
 	"sync"
+	"testing"
 
 	"go.etcd.io/etcd/raft/v3/raftpb"
 
 	"github.com/matrixorigin/matrixcube/pb/meta"
+	"github.com/matrixorigin/matrixcube/util/leaktest"
+	"github.com/matrixorigin/matrixcube/vfs"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func getNodeInfo(shardID uint64, replicaID uint64) nodeInfo {
@@ -178,4 +184,32 @@ func (h *testMessageHandler) getMessageCount(m map[nodeInfo]uint64,
 		return v
 	}
 	return 0
+}
+
+func TestContainerResolverReturnEmptyAddr(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	fs := vfs.GetTestFS()
+	defer vfs.ReportLeakedFD(fs, t)
+
+	trans := NewTransport(nil, testTransportAddr, 2,
+		nil, nil, nil,
+		getTestSnapshotDir, func(storeID uint64) (string, error) { return "", nil }, fs)
+	require.NoError(t, trans.Start())
+	defer trans.Close()
+
+	assert.False(t, trans.Send(meta.RaftMessage{}))
+}
+
+func TestContainerResolverReturnError(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	fs := vfs.GetTestFS()
+	defer vfs.ReportLeakedFD(fs, t)
+
+	trans := NewTransport(nil, testTransportAddr, 2,
+		nil, nil, nil,
+		getTestSnapshotDir, func(storeID uint64) (string, error) { return "", errors.New("error") }, fs)
+	require.NoError(t, trans.Start())
+	defer trans.Close()
+
+	assert.False(t, trans.Send(meta.RaftMessage{}))
 }

